src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsServerRouter/DiagnosticsServerRouterFactory.cs

```diff
@@ -333,13 +332,16 @@ public virtual async Task<Stream> ConnectTcpStreamAsync(CancellationToken token)
 
             connectTimeoutTokenSource.CancelAfter(TcpClientTimeoutMs);
 
+            if (!retry && _auto_shutdown)
+                retry = true;
```

Comment:
Why does `_auto_shutdown` override the method parameter `retry`?

Reply:
The auto shutdown requires retries until timeout, but I realize that instead of using a default value for retry, we should be split into two functions instead, where a call not specifying retry should use auto shutdown as its retry value. Will adjust.